### ISSUE TITLE
ncat win32 in listen mode terminates immediately after an 'empty' client connects

### ISSUE THREAD

**Issue by Des-Nerger** [nmap/nmap#104]
```
win32.machine> ncat -l
linux.machine> ncat win32.machine
win32.machine> close: No error
linux.machine> Ncat: Connection reset by peer.
linux.machine> echo $?
linux.machine> 1
```
At reverse order (linux listens, windows connects) ncat works fine (i.e. initiates character exchanging until first EOF). Also:
```
win32.machine> ncat -l
linux.machine> nc.openbsd win32.machine 31337
win32.machine> close: No error
linux.machine> echo $?
linux.machine> 0
```

**Comment by dmiller-nmap** [comment]
@Des-Nerger Will you please provide output of these commands when using the `-vvv` option for maximum debugging? Thanks!

**Comment by Des-Nerger** [comment]
http://pastebin.com/2SeHG3Uf ("Адрес несовместим с выбранным протоколом" means "The address is incompatible with the selected protocol")
http://pastebin.com/agX2n8NT

**Comment by dmiller-nmap** [comment]
@Des-Nerger I notice you are using the default port for Ncat, 31337/tcp. Because of the association of this port ("elite") with hacking, some security software and antivirus can interfere with communications. Do you get the same problem if you specify a different port?

    C:\> ncat -vvv -l -p 9999

    sh-$ ncat -vvv win32.machine 9999

**Comment by Des-Nerger** [comment]
Yes, I get the same problem after doing that. And I don't have any security software enabled

**Comment by Des-Nerger** [comment]
Appending `-4` to the parameter list of the listening ncat has helped.
//Hmm, there is no `close: No error` after a connection now. But this is nothing.

**Comment by udhgee** [comment]
Thanks Des-Nerger, appending -4 worked for me too.

**Comment by dmiller-nmap** [comment]
`-4` forces IPv4-only. By default, Ncat listens on both IPv4 and IPv6. Can you tell us more about your setup and why IPv6 might not be working? Specifically, I would like to know:

1. Version of Windows
2. output of `nmap --iflist` or `ipconfig /all`
3. output of `netsh interface ipv6 show interfaces`
4. Was this installed from our executable installer, from the zipfile, or built by you?
5. Does the problem persist with Nmap/Ncat 7.01?

**Comment by fyodor** [comment]
We're closing this since we never got feedback from OP to @dmiller-nmap's latest questions and it's also not clear that any such issue that may have existed more than 10 years ago still does.  Folks are welcome to file a new issue if they see anything similar with modern Nmap.
